config/permissions.py

```diff
@@ -0,0 +1,193 @@
+from enum import Enum, auto, unique
+from config.exceptions import PermissionDeniedException
+from apps.user.models import User
+
+
+class BasePermissions():
+
+    # ------------ Define this after using this as base -----------
+    @unique
+    class Permission(Enum):
+        pass
+    __error_message__ = {}
+    PERMISSION_MAP = {}
+    CONTEXT_PERMISSION_ATTR = ''
+    # ------------ Define this after using this as base -----------
+
+    DEFAULT_PERMISSION_DENIED_MESSAGE = PermissionDeniedException.default_message
+
+    @classmethod
+    def get_permissions(cls, role):
+        return cls.PERMISSION_MAP.get(role) or []
+
+    @classmethod
+    def get_permission_message(cls, permission):
+        return cls.__error_message__.get(permission, cls.DEFAULT_PERMISSION_DENIED_MESSAGE)
+
+    @classmethod
+    def check_permission(cls, info, *perms):
+        permissions = getattr(info.context, cls.CONTEXT_PERMISSION_ATTR)
+        if permissions:
+            return all([perm in permissions for perm in perms])
+
+    @classmethod
+    def check_permission_from_serializer(cls, context, *perms):
+        permissions = getattr(context, cls.CONTEXT_PERMISSION_ATTR)
+        if permissions:
+            return all([perm in permissions for perm in perms])
+
+
+class BookPermissions(BasePermissions):
```

Comment:
Do we need to define for each object? Can't we do it all in one? I will recommend a single permission class.